src/main/java/seedu/address/model/entry/Link.java

```diff
@@ -1,46 +1,38 @@
 package seedu.address.model.entry;
 
 import static java.util.Objects.requireNonNull;
-import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.net.MalformedURLException;
+import java.net.URL;
 
 /**
  * Represents a Entry's link in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidConstructionLink(String)}
```

Comment:
We delete ALL checks from this file, and rely solely on MalformedURLException to ensure every Link is valid.